cloudinit/sources/DataSourceAzure.py

```diff
@@ -1345,12 +1345,21 @@ def parse_network_config(imds_metadata):
                 for addr6 in intf['ipv6']['ipAddress']:
                     privateIpv6 = addr6['privateIpAddress']
                     if privateIpv6:
-                        dev_config['dhcp6'] = True
-                        # non-primary interfaces should have a higher
-                        # route-metric (cost) so default routes prefer
-                        # primary nic due to lower route-metric value
-                        dev_config['dhcp6-overrides'] = dhcp_override
-                        break
+                        if dev_config.get('dhcp6', False):
```

Comment:
What do you think about enumeration so we know we're on the primary interface.

Also,  if we are on a single interface, we need to dhcp6 *and* also assign ipv6 static addresses.

netplan already supports this:

```
ethernets:
   eth0:
      dhcp6: True
      addresses: [addr6, addr6, addr6]

```

Reply:
I'm not sure I understand your comment here. Is the example netplan for a nic with 4 total ipv6 addresses listed by IMDS or 3 total?

Currently for a nic with 2 ipv4 and 3 ipv6 addrs in IMDS you'd get

ethernets:
  eth0:
   dhcp6: True
   dhcp4: True
   addresses: [addr4#2/prefix4, addr6#2/prefix6, addr6#3/prefix6, addr6#4/prefix6]

Reply:
Sorry, it was incomplete, I just meant that we can extract/combine all of the v4/v6 static ips beyond the first IP on each of v4/v6  which obtain IP via DHCP.

The code you pushed does exactly what I was thinking.